pkg/ovs/openflow/interfaces.go

```diff
@@ -103,9 +103,7 @@ type Bridge interface {
 	GetTableByID(id uint8) (Table, error)
 	CreateGroupTypeAll(id GroupIDType) Group
 	CreateGroup(id GroupIDType) Group
-	DeleteGroup(id GroupIDType) error
```

Comment:
It's a bit surprising to have `CreateGroup*` but not `DeleteGroup`. Why do we still need `CreateGroup`?

Reply:
We need to use OpenFlow group for Service/Multicast scenario, `CreateGroup` is called to generate an OpenFlow group resource. Originally, we call `DeleteGroup` to remove the group resource from the OFSwitch.groupDb ( ofnet logic). In this change, we have modified the logic to not save the group/table/meter in ofnet, so we don't need to remove resources from OFSwitch.groupDb. That is why I remove this `DeleteGroup`. As for OF group deletion from OVS, it is called with bundle message in ofClient, e.g., https://github.com/antrea-io/antrea/blob/main/pkg/agent/openflow/client.go#L694.

Reply:
Is there a reason why we don't also use a bundle message for group creation?

Reply:
antrea uses `ofctrl.Group` to maintain the OF group configurations including the buckets, so we firstly need to create such an object. Then antrea uses bundle message to install the group resource on OVS, `ofctrl.Group` has provided the API to translate the resource to a GroupModification message.

Reply:
I see, so this is actually more like a builder or a constructor. They just create a Go object.

Reply:
Perhaps `NewGroup` is more proper?

Reply:
`NewGroup` is taken in ofctrl which expects to create object and add it into the groupDb. I don't want to modify the existing function name

Reply:
I was thinking of `NewGroup` too, but I think it's confusing to have a `New` function as part of an interface.

What's a bit misleading right now is that the interface combines methods that just create Go objects (e.g., `CreateGroup`) with methods that generate OpenFlow messages (`DeleteFlowsByCookie` / `AddOFEntriesInBundle` / `AddFlowsInBundle` / ...). It's a bit hard to know what the method will do based on the name alone. I wonder if we could improve that, maybe by splitting the interface.

Reply:
How about `GenerateGroup'?

Reply:
> It's a bit hard to know what the method will do based on the name alone. I wonder if we could improve that, maybe by splitting the interface.

@antoninbas, I am not very clear on your thoughts about how to split the interface, what parts do you want to split?

Reply:
I think you are meaning to use the new name style for functions in antrea not in ofnet, right? I can update the existing `Create*` to `Make*`. (`Build*` is not preferred as we always use it to generate a "*Builder", e.g., "FlowBuilder", "BucketBuilder")

@tnqn any idea on the new name for the APIs ?

Reply:
@antoninbas I feel `Bridge.NewGroup` is fine. I saw several such cases in Go and K8s code. When `A` has a method named `NewB`, it means constructing an instance of `B` from `A`. For example:

https://github.com/golang/go/blob/fd13444b2b4bf1ce6b05980b1b139402048f0363/src/net/http/server.go#L623-L633
https://github.com/kubernetes/kubernetes/blob/af99df6a6bead927d61746dc431fc1b526af55a0/staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go#L334

@wenyingd In ofnet, I think you shouldn't make the constructor of `Group` a method of `OFSwitch` and name it `CreateGroup` in https://github.com/antrea-io/ofnet/pull/59/files#diff-3c6f0d72cfad8ce2c810076de9801d7bbf22ac2bfb08c252549bc9dd38b9adb1R145.
```
func (self *OFSwitch) CreateGroup(groupId uint32, groupType GroupType) *Group {
	return &Group{
		ID:        groupId,
		GroupType: groupType,
		Switch:    self,
	}
}
```

It is a very typical constructor, and by convention it should be like:
```
func NewGroup(id uint32, groupType GroupType, ofSwitch *OFSwitch) *Group {
	return &Group{
		ID:        id,
		GroupType: groupType,
		Switch:    ofSwitch,
	}
}
```

Then it could just be:
```
func (b *OFBridge) newGroupWithType(id GroupIDType, groupType ofctrl.GroupType) Group {
	ofctrlGroup := ofctrl.NewGroup(uint32(id), groupType, b.ofSwitch)
	g := &ofGroup{bridge: b, ofctrl: ofctrlGroup}
	return g
}
```

Reply:
updated accordingly.